pkg/gitreceive/k8s_util.go

```diff
@@ -0,0 +1,163 @@
+package gitreceive
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pborman/uuid"
+	"k8s.io/kubernetes/pkg/api"
+	apierrs "k8s.io/kubernetes/pkg/api/errors"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+func dockerBuilderPodName(appName, shortSha string) string {
+	uid := uuid.New()[:8]
+	return fmt.Sprintf("dockerbuild-%s-%s-%s", appName, shortSha, uid)
+}
+
+func slugBuilderPodName(appName, shortSha string) string {
+	uid := uuid.New()[:8]
+	return fmt.Sprintf("slugbuild-%s-%s-%s", appName, shortSha, uid)
+}
+
+func dockerBuilderPod(debug, withAuth bool, name, namespace string, env map[string]interface{}, tarURL, imageName string) *api.Pod {
+	pod := buildPod(debug, withAuth, name, namespace, env)
+
+	pod.Spec.Containers[0].Name = "deis-dockerbuilder"
+	pod.Spec.Containers[0].Image = "quay.io/deisci/dockerbuilder:v2-beta"
```

Comment:
Note to others: a later PR can make this image configurable, and can hence solve https://github.com/deis/builder/issues/66